src/operator/buffer.ts

```diff
@@ -23,6 +23,10 @@ export function buffer<T>(closingNotifier: Observable<any>): Observable<T[]> {
   return this.lift(new BufferOperator<T>(closingNotifier));
 }
 
+export interface BufferSignature<T> {
+  (closingNotifier: Observable<any>): Observable<T[]>;
```

Comment:
`closingNotifier` calls through to `subscribeToResult`, so technically this could be `ObservableInput<T>`.  That doesn't really make sense for the more finite values from `Promise<T>`, `Iterable<T>` or `Array<T>`.  So I think the best option is to explicitly leave these as just `Observable<T>`.